mosaic-runtime/src/commonTest/kotlin/com/jakewharton/mosaic/layout/PaddingTest.kt

```diff
@@ -0,0 +1,313 @@
+		)
+	}
+
+	@Test fun paddingLeftBottomDebug() {
+		val actual = Modifier.padding(left = 1, bottom = 2).toString()
+		assertThat(actual).isEqualTo("Padding(l=1, t=0, r=0, b=2)")
+	}
+
+	@Test fun paddingAllDebug() {
+		val actual = Modifier.padding(2).toString()
+		assertThat(actual).isEqualTo("Padding(2)")
+	}
+
+	@Test fun paddingHorizontalDebug() {
+		val actual = Modifier.padding(horizontal = 2).toString()
+		assertThat(actual).isEqualTo("Padding(h=2, v=0)")
+	}
+
+	@Test fun paddingVerticalDebug() {
+		val actual = Modifier.padding(vertical = 2).toString()
+		assertThat(actual).isEqualTo("Padding(h=0, v=2)")
+	}
+
+	@Test fun intrinsicMeasurements() {
+		val padding = 100
+
+		var error: Throwable? = null
+		testIntrinsics(
+			@Composable {
+				TestBox(modifier = Modifier.padding(padding)) {
+					Container(Modifier.aspectRatio(2f)) { }
+				}
+			},
+		) { minIntrinsicWidth, minIntrinsicHeight, maxIntrinsicWidth, maxIntrinsicHeight ->
+			// Spacing is applied on both sides of an axis
+			val totalAxisSpacing = padding * 2
+
+			// When the width/height is measured as 3 x the padding
+			val testDimension = padding * 3
+			// The actual dimension for the AspectRatio will be: test dimension - total padding
+			val actualAspectRatioDimension = testDimension - totalAxisSpacing
+
+			// When we measure the width first, the height will be half
+			val expectedAspectRatioHeight = actualAspectRatioDimension / 2f
+			// When we measure the height first, the width will be double
+			val expectedAspectRatioWidth = actualAspectRatioDimension * 2
+
+			// Add back the padding on both sides to get the total expected height
+			val expectedTotalHeight = (expectedAspectRatioHeight + totalAxisSpacing).toInt()
+			// Add back the padding on both sides to get the total expected height
+			val expectedTotalWidth = expectedAspectRatioWidth + totalAxisSpacing
+
+			try {
+				// Min width
+				assertThat(totalAxisSpacing).isEqualTo(minIntrinsicWidth(0))
+				assertThat(expectedTotalWidth).isEqualTo(minIntrinsicWidth(testDimension))
+				assertThat(totalAxisSpacing).isEqualTo(minIntrinsicWidth(Constraints.Infinity))
+				// Min height
+				assertThat(totalAxisSpacing).isEqualTo(minIntrinsicHeight(0))
+				assertThat(expectedTotalHeight).isEqualTo(minIntrinsicHeight(testDimension))
+				assertThat(totalAxisSpacing).isEqualTo(minIntrinsicHeight(Constraints.Infinity))
+				// Max width
+				assertThat(totalAxisSpacing).isEqualTo(maxIntrinsicWidth(0))
+				assertThat(expectedTotalWidth).isEqualTo(maxIntrinsicWidth(testDimension))
+				assertThat(totalAxisSpacing).isEqualTo(maxIntrinsicWidth(Constraints.Infinity))
+				// Max height
+				assertThat(totalAxisSpacing).isEqualTo(maxIntrinsicHeight(0))
+				assertThat(expectedTotalHeight).isEqualTo(maxIntrinsicHeight(testDimension))
+				assertThat(totalAxisSpacing).isEqualTo(maxIntrinsicHeight(Constraints.Infinity))
+			} catch (t: Throwable) {
+				error = t
+			}
+		}
+
+		error?.let { throw it }
```

Comment:
Seems like this behavior should be added directly to `testIntrinsics` helper

Reply:
I looked and checked without the `error` variable at all, it works without it